Userland/Libraries/LibJS/Runtime/MathObject.cpp

```diff
@@ -498,9 +498,22 @@ JS_DEFINE_NATIVE_FUNCTION(MathObject::sinh)
 // 21.3.2.13 Math.cosh ( x ), https://tc39.es/ecma262/#sec-math.cosh
 JS_DEFINE_NATIVE_FUNCTION(MathObject::cosh)
 {
+    // 1. Let n be ? ToNumber(x).
     auto number = TRY(vm.argument(0).to_number(global_object));
+
+    // 2. If n is NaN, return NaN.
     if (number.is_nan())
         return js_nan();
+
+    // 3. If n is +∞𝔽 or n is -∞𝔽, return +∞𝔽.
+    if (number.is_positive_infinity() || number.is_negative_infinity())
+        return js_infinity();
+
+    // 4. If n is +0𝔽 or n is -0𝔽, return 1𝔽.
+    if (number.is_positive_zero() || number.is_negative_zero())
+        return Value(1);
+
+    // 5. Return an implementation-approximated Number value representing the result of the hyperbolic cosine of ℝ(n).
```

Comment:
This behavior is actually mandated by the [POSIX standard](https://pubs.opengroup.org/onlinepubs/9699919799/functions/cosh.html). The redundant checks here obviously don't hurt, but ideally, LibM should be fixed as well.

cc @Hendiadyoin1

Reply:
Yeah I agree that is probably the best fix, from a js perspective we don't know if ::cosh is POSIX compliant so I think we can keep the checks here.

If you want to test this try running without virtualization then you get this incorrect results for infinity etc.